### PR TITLE
[fix](cloud) Fix cloud create table need cluster auth and no cluster …

### PR DESCRIPTION
**Pull request by deardeng**
…auth select throw npe

```
2024-03-04 15:24:22,555 DEBUG (mysql-nio-pool-9|574) [Role.checkCloudPriv():411] failed to get wanted privs: priv predicate: OR, Admin_priv Usage_priv Cluster_Usage_priv , granted:
2024-03-04 15:24:22,555 WARN (mysql-nio-pool-9|574) [ConnectContext.getCloudCluster():1094] cant get a valid cluster for user 'nereids_user'@'%' to use
2024-03-04 15:24:22,555 DEBUG (mysql-nio-pool-9|574) [CloudCoordinator.prepare():67] get cluster by context null
2024-03-04 15:24:22,555 WARN (mysql-nio-pool-9|574) [CloudCoordinator.prepare():75] invalid clusterName: null
2024-03-04 15:24:22,555 WARN (mysql-nio-pool-9|574) [StmtExecutor.sendResult():1700] cancel fragment query_id:f4e0832dc2a94f0c-bba7afaee9bdb38b cause null
2024-03-04 15:24:22,555 WARN (mysql-nio-pool-9|574) [Coordinator.cancel():1473] Cancel execution of query f4e0832dc2a94f0c-bba7afaee9bdb38b, this is a outside invoke
2024-03-04 15:24:22,555 INFO (mysql-nio-pool-9|574) [QueryQueue.returnToken():154] wgId= 1, version=0,maxConcurrency=2147483647, maxQueueSize=0, queueTimeout=0, currentRunningQueryNum=5, currentWaitingQueryNum=0
2024-03-04 15:24:22,555 DEBUG (mysql-nio-pool-9|574) [QeProcessorImpl.unregisterQuery():145] Deregister query id f4e0832dc2a94f0c-bba7afaee9bdb38b
2024-03-04 15:24:22,555 WARN (mysql-nio-pool-9|574) [ConnectProcessor.handleQueryException():361] Process one query failed because unknown reason:
java.lang.NullPointerException: null
        at org.apache.doris.qe.Coordinator.computeScanRangeAssignment(Coordinator.java:2280) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.Coordinator.processFragmentAssignmentAndParams(Coordinator.java:578) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.cloud.qe.CloudCoordinator.processFragmentAssignmentAndParams(CloudCoordinator.java:84) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.Coordinator.execInternal(Coordinator.java:651) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.Coordinator.exec(Coordinator.java:630) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.sendResult(StmtExecutor.java:1589) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.handleQueryStmt(StmtExecutor.java:1562) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.handleQueryWithRetry(StmtExecutor.java:711) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.executeByNereids(StmtExecutor.java:663) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:496) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:475) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.executeQuery(ConnectProcessor.java:279) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:197) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.MysqlConnectProcessor.handleQuery(MysqlConnectProcessor.java:176) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.MysqlConnectProcessor.dispatch(MysqlConnectProcessor.java:205) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.MysqlConnectProcessor.processOnce(MysqlConnectProcessor.java:258) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.mysql.ReadListener.lambda$handleEvent$0(ReadListener.java:52) ~[doris-fe.jar:1.2-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_131]
```

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

File: regression-test/suites/account_p0/test_nereids_authentication.groovy [hunk 1]
```diff
@@ -26,6 +26,11 @@ suite("test_nereids_authentication", "query") {
             PROPERTIES ('replication_num' = '1')
         """
     }
+    def isCloudMode = {
```
Comment (yujun777) [inline regression-test/suites/account_p0/test_nereids_authentication.groovy:29]: use Suite.isCloudCluster() ?

Comment (yujun777) [reply regression-test/suites/account_p0/test_nereids_authentication.groovy:29]: maybe change Suite.isCloudCluster() to Suite.isCloudMode() is better

Comment (deardeng) [reply regression-test/suites/account_p0/test_nereids_authentication.groovy:29]: Suite.isCloudCluster() 是判断配置文件是否配置了cloud_unique_id，依赖配置文件。

我感觉我自己def这个方法，更好点可以自适应。show backends，然后看tag里面是否包含cloud的信息。

File: fe/fe-core/src/main/java/org/apache/doris/cloud/qe/CloudCoordinator.java [hunk 2]
```diff
@@ -81,13 +92,73 @@ protected void prepare() {
     }
 
     protected void processFragmentAssignmentAndParams() throws Exception {
-        super.processFragmentAssignmentAndParams();
+        prepare();
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/qe/CloudCoordinator.java:95]: override prepare  and move line 96 ~ 100 at the end of CloudCoordinator.end

Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/qe/CloudCoordinator.java:95]: delelte override processFragmentAssignmentAndParams

**Review by yujun777 — COMMENTED** [review]
need some update

File: fe/fe-core/src/main/java/org/apache/doris/cloud/qe/CloudCoordinator.java (hunk 2 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/qe/CloudCoordinator.java]: fixed

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java [hunk 3]
```diff
@@ -601,6 +601,23 @@ public void addReplica(long tabletId, Replica replica) {
         }
     }
 
+    // cloud mode, create table not need backendId, represent with -1.
+    public void addCloudReplica(long tabletId, Replica replica) {
+        long stamp = writeLock();
+        try {
+            Preconditions.checkState(tabletMetaMap.containsKey(tabletId),
+                    "cloudTablet " + tabletId + " not exists, replica " + replica.getId());
+            replicaMetaTable.put(tabletId, (long) -1, replica);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java:610]: try to use the original`addReplica` to reduce duplicated code?
```suggestion
            long bid = isCloudMode() ? -1 : replica.getBackendId();
            replicaMetaTable.put(tabletId, bid, replica);
```

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java (hunk 3 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java]: fix

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**